Extensions/soft_refresh.js

```diff
@@ -11,212 +11,184 @@ XKit.extensions.soft_refresh = new Object({
-			$(document).on("click", "#home_button", XKit.extensions.soft_refresh.logo_clicked);
-			// Need to change all children to make sure the user doesn't click the new posts count number.
-			$("#home_button").children().attr('href', '#');
+			$("#home_button > a")
+				.attr("data-old-href", $("#home_button > a").attr("href"))
+				.removeAttr("href")
+				.click(this.clicked);
 		}
+	},
 
-		XKit.extensions.soft_refresh.do_post_ids();
-
+	clicked: function(event) {
+		if (event.altKey) {
+			location.reload(true);
+		} else {
+			XKit.extensions.soft_refresh.start_load();
+		}
 	},
 
-	do_post_ids: function() {
+	start_load: function() {
+		if (this.loading) { return; }
+		this.loading = true;
 
-		$(".posts .post").each(function() {
+		$("html, body").animate({ scrollTop: 0 }, "slow");
 
-			$(this).parent().attr('data-xkit-post-id', $(this).attr('data-post-id'));
+		$("#new_post").after('<div id="xkit_soft_refresh">Checking for new posts</div>');
+		$("#xkit_soft_refresh").slideDown("fast");
 
+		$("#new_post_notice_container").css({
+			transform: "scale(0)",
+			opacity: 0
 		});
 
+		this.top_post = $("#posts > li.post_container:not(#new_post_buttons)").first();
+		this.request("");
 	},
 
-	logo_clicked: function(event) {
-
-		if (event.altKey) {
-			location.reload(true);
-			return;
-		}
-		XKit.extensions.soft_refresh.load_posts();
-
-	},
-
-	load_posts: function() {
-
-		if (typeof XKit.extensions.filter_by_type !== "undefined") {
-			if (XKit.extensions.filter_by_type.running === true) {
-				if (XKit.extensions.filter_by_type.current_filter !== "0") {
-					location.reload(true);
+	request: function(page) {
+		fetch(`https://www.tumblr.com/dashboard${page}`)
+			.then(response => {
+				if (!response.ok) {
+					this.show_error();
 					return;
 				}
-			}
-		}
 
-		if (this.loading === true) { return; }
-		this.loading = true;
-
-		$("#new_post").after("<div id=\"xkit_soft_refresh\">Checking for new posts</div>");
-		$("#xkit_soft_refresh").slideDown('fast');
-
-		document.title = this.default_page_title;
-		$("#new_post_notice_container").css("display", "none");
-		$("#new_post_notice_container .tab_notice_value").html("0");
-
-		function soft_refresh_hit_triggers() {
-			var postIds = add_tag;
-			var Tumblr = window.Tumblr || window.top.Tumblr;
-
-			try {
-				postIds.forEach(function(id) {
-					var postId = "post_" + id;
-					var postElement = jQuery(document.getElementById(postId));
-					var fakeView = new Tumblr.PostView({
-						el: postElement,
-						model: Tumblr.PostsView.prototype.createPostModelFromEl(postElement)
-					});
-					Tumblr.Events.trigger("postsView:createPost", fakeView);
-				});
-			} catch (e) {
-				console.warn("soft_refresh new approach failed:", e);
-			}
-			Tumblr.Events.trigger("posts:load");
-			Tumblr.Events.trigger("DOMEventor:updateRect");
-			Tumblr.AudioPlayer.update_all();
-		}
+				let end = false;
+
+				response.text().then(responseText => {
+					if (page === "") {
```

Comment:
if we're loading a ton of posts, the value might update while we're loading posts, since we fetch the first page of the dashboard first and then all the older ones until we get to a post we've already got